src/Serilog.Extensions.Logging/Extensions/Logging/SerilogLogger.cs

```
@@ -153,6 +155,7 @@ void Write<TState>(LogEventLevel level, EventId eventId, TState state, Exception
 
         var parsedTemplate = MessageTemplateParser.Parse(messageTemplate ?? "");
         var evt = new LogEvent(DateTimeOffset.Now, level, exception, parsedTemplate, properties);
+        parsed = true;
```

should parsed be after the write?

---

No. The whole point is not to swallow any exceptions if the control is transferred inside `logger.Write` because Serilog core code takes care of exception handling. It swallows exceptions in case of `WriteTo` and it throws them back to the caller in case of `AuditTo`.

---

thanks for the clarification